### ISSUE TITLE
Additional incorrect clashing_extern_decl warning for `#[repr(transparent)]` non-zero types

### ISSUE THREAD

**Issue by jumbatm** [rust-lang/rust#75739]
It seems that there is a case still not being fixed, when `#[repr(transparent)]` is used with non-zero types.
```rust
mod a {
    use std::num::NonZeroUsize;
    extern "C" {
        fn a() -> NonZeroUsize;
    }
}

mod b {
    #[repr(transparent)]
    struct X(NonZeroUsize);
    use std::num::NonZeroUsize;
    extern "C" {
        fn a() -> X;
    }
}
```

_Originally posted by @nbdd0121 in https://github.com/rust-lang/rust/issues/73735#issuecomment-677152928_
